src/cryptoadvance/specter/templates/includes/notifications.js

```diff
@@ -0,0 +1,377 @@
+{% include "includes/message_box.js" %}
+
+
+var userToken = null;
+
+/**
+ * creating a notification from JS
+ * 
+ * Example:
+ * createNotification('this is the title', {target_uis:['js_message_box', 'webapi'], body:'body line 1\nline 2', image:'/static/img/ghost_3d.png', timeout:3000})
+ */ 
+ async function createNotification(title, options){ 
+    if (!websocket){return}
+    if (websocket.readyState === WebSocket.OPEN) {
+        websocket.send(JSON.stringify( {'user_token': userToken, 'title':title, 'options': options}));
+    } else {
+        // If the socket is not open yet, retry in 1 s
+        setTimeout(createNotification, 1000, title, options);  
+    }
+}
+
+
+
+function onClose(id, targetUi){
+    //console.log('closed message')    
+    createNotification('on_close', {'target_uis':['internal_notification'], 'data':{'id':id, 'target_ui':targetUi}});
+}
+
+
+function onShow(id, targetUi, success=true){
+    //console.log('closed message')
+    createNotification('on_show', {'target_uis':['internal_notification'], 'data':{'id':id, 'target_ui':targetUi, 'success':success}});
+}
+
+
+function setTargetUiAvailability(targetUi, isAvailable, id=null) {
+    createNotification('set_target_ui_availability', {'target_uis':['internal_notification'], 'data':{'id':id, 'target_ui':targetUi, 'is_available':isAvailable}});
+}
+
+
+
+
+    /* jsNotification structure see https://notifications.spec.whatwg.org/#api
+    options : {  
```

Comment:
Ups, are those the options? But we're not supporting all of them, do we?

Reply:
Better documentation in  950cc4ff